libpod/stats.go

```diff
@@ -0,0 +1,126 @@
+package libpod
+
+import (
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/containerd/cgroups"
+	"github.com/opencontainers/runc/libcontainer"
+	"github.com/pkg/errors"
+)
+
+// ContainerStats contains the statistics information for a running container
+type ContainerStats struct {
+	Container   string
+	CPU         float64
+	cpuNano     uint64
+	systemNano  uint64
+	MemUsage    uint64
+	MemLimit    uint64
+	MemPerc     float64
+	NetInput    uint64
+	NetOutput   uint64
+	BlockInput  uint64
+	BlockOutput uint64
+	PIDs        uint64
+}
+
+// GetContainerStats gets the running stats for a given container
+func (c *Container) GetContainerStats(previousStats *ContainerStats) (*ContainerStats, error) {
+
+	stats := new(ContainerStats)
+	cgroup, err := cgroups.Load(cgroups.V1, c.CGroupPath())
+	if err != nil {
+		return stats, errors.Wrapf(err, "unable to load cgroup at %+v", c.CGroupPath())
+	}
+
+	cgroupStats, err := cgroup.Stat()
+	if err != nil {
+		return stats, errors.Wrapf(err, "unable to obtain stat")
```

Comment:
Nit: can you make the error message "cgroup stat" instead of just stat? A little more clear that way

Reply:
done